### PR TITLE
feat: display branch selection in the tree node [IDE-446]

### PR DESCRIPTION
**Pull request by acke**
### Description

This pr covers the UI parts of selecting a base branch for delta findings. No actually calls to language server for available local branches, or to set the choose branch is a part of this pr.

This PR handles tree view for Security Code and Code Quality depending on the Delta Findings feature flag. When enabled, the title for the tree node shows

- Code Security - New Issues
- Code Quality - New Issues

### Checklist

- [ ] Tests added and all succeed
- [ ] Linted
- [ ] CHANGELOG.md updated
- [ ] README.md updated, if user-facing

### Screenshots / GIFs

https://github.com/user-attachments/assets/9fe4ede8-c2bf-481d-8698-46decb49c096

### CODE REVIEW

File: src/snyk/common/commands/commandController.ts [hunk 1]
```diff
@@ -60,6 +61,10 @@ export class CommandController {
     await this.executeCommand(SNYK_SET_TOKEN_COMMAND, this.authService.setToken.bind(this.authService));
   }
 
+  async setBaseBranch(): Promise<void> {
+    await this.executeCommand(SNYK_SET_BASE_BRANCH_COMMAND, this.authService.setBaseBranch.bind(this.authService));
+  }
```
Comment (bastiandoetsch) [inline src/snyk/common/commands/commandController.ts:66]: this will probably not stay like this, but use a setting

**Review by bastiandoetsch — CHANGES_REQUESTED** [review]
I think this needs to wait until the LS side is implemented for correct integration. Correct me, if there's another story that handles the LS integration.

File: src/snyk/common/views/issueTreeProvider.ts [hunk 2]
```diff
@@ -275,7 +294,10 @@ export abstract class ProductIssueTreeProvider<T> extends AnalysisTreeNodeProvid
   }
 
   protected getIssueFoundText(nIssues: number, _: number): string {
-    return `Snyk found ${!nIssues ? 'no issues! ✅' : `${nIssues} issue${nIssues === 1 ? '' : 's'}`}`;
+    if (!nIssues) {
+      return '✅ Congrats! No issues found!';
+    }
+    return `Snyk found ${nIssues} issue${nIssues === 1 ? '' : 's'}`;
```
Comment (cat2608) [inline src/snyk/common/views/issueTreeProvider.ts:300]: Small refactor for consistency. Agreed with Arvrid and Neil 🙌 

|Before|After|
|-|-|
|before|after|

File: package.json [hunk 3]
```diff
@@ -310,12 +320,12 @@
...
         {
           "command": "snyk.start",
-          "when": "view == 'snyk.views.analysis.code.security' || view == 'snyk.views.analysis.code.quality' || view == 'snyk.views.analysis.oss' || view == 'snyk.views.analysis.configuration'",
+          "when": "view == 'snyk.views.analysis.code.security' || view == 'snyk.views.analysis.code.security.delta' || view == 'snyk.views.analysis.code.quality' || view == 'snyk.views.analysis.code.quality.delta' || view == 'snyk.views.analysis.oss' || view == 'snyk.views.analysis.configuration'",
           "group": "navigation"
         },
         {
           "command": "snyk.settings",
-          "when": "view == 'snyk.views.analysis.code.security' || view == 'snyk.views.analysis.code.quality' || view == 'snyk.views.analysis.oss' || view == 'snyk.views.welcome' || view == 'snyk.views.analysis.configuration'",
+          "when": "view == 'snyk.views.analysis.code.security' || view == 'snyk.views.analysis.code.security.delta' || view == 'snyk.views.analysis.code.quality' || view == 'snyk.views.analysis.code.quality.delta' || view == 'snyk.views.analysis.oss' || view == 'snyk.views.welcome' || view == 'snyk.views.analysis.configuration'",
```
Comment (cat2608) [inline package.json:328]: This adds the gear ⚙️ and play ▶️ buttons on the right of the title

File: src/snyk/snykCode/views/securityIssueTreeProvider.ts [hunk 4]
```diff
@@ -42,18 +42,20 @@ export default class CodeSecurityIssueTreeProvider extends IssueTreeProvider {
   protected getIssueFoundText(nIssues: number, ignoredIssueCount: number): string {
     if (nIssues > 0) {
       let text;
+
       if (nIssues === 1) {
         text = `${nIssues} vulnerability found by Snyk`;
```
Comment (bastiandoetsch) [inline src/snyk/snykCode/views/securityIssueTreeProvider.ts:47]: Shouldn't this be issues, too?

File: src/snyk/snykCode/views/securityIssueTreeProvider.ts [hunk 5]
```diff
@@ -42,18 +42,20 @@ export default class CodeSecurityIssueTreeProvider extends IssueTreeProvider {
...
       } else {
         text = `✋ ${nIssues} vulnerabilities found by Snyk`;
```
Comment (bastiandoetsch) [inline src/snyk/snykCode/views/securityIssueTreeProvider.ts:49]:
```suggestion
        text = `✋ ${nIssues} issues found by Snyk`;
```

File: src/snyk/common/views/issueTreeProvider.ts [hunk 6]
```diff
@@ -109,6 +115,19 @@ export abstract class ProductIssueTreeProvider<T> extends AnalysisTreeNodeProvid
     return nodes;
   }
 
+  getBaseBranch(): TreeNode | null {
+    const deltaFindingsEnabled = this.configuration.getDeltaFindingsEnabled();
+
+    //TODO: get the actual base branch from Snyk Language Server
```
Comment (bastiandoetsch) [inline src/snyk/common/views/issueTreeProvider.ts:121]: This should just ask a config setting, what's the base branch for the folder. LS sends a folderConfig notification for all known workspace folders when they are registered at LS.

**Review by bastiandoetsch — APPROVED** [review]

**Review by ShawkyZ — APPROVED** [review]

**Pull request merged**